packages/core/datadog-sourcemaps.gradle

```diff
@@ -148,6 +176,76 @@ private def getDatadogCiExecPath(reactConfig) {
     return defaultPath
 }
 
+/**
+ * We use a function here to resolve the Datadog React Native package path.
+ * If DATADOG_REACT_NATIVE_PATH env variable is defined, it will be returned (if valid).
+ */
+private def getDatadogReactNativePath(File workingDirectory) {
+    def defaultPath = "${workingDirectory.absolutePath}/node_modules/@datadog/mobile-react-native"
+
+    // Try to retrieve the path from ENV variable
+    def envPath = System.getenv('DATADOG_REACT_NATIVE_PATH')
+    if (envPath != null) {
+        if (isValidDatadogPackagePath(envPath)) {
+            return envPath
+        } else {
+            println("WARNING: Ignoring DATADOG_REACT_NATIVE_PATH as it does not point to a valid @datadog/mobile-react-native package")
+        }
+    }
+
+    def nodeExecutable = Os.isFamily(Os.FAMILY_WINDOWS) ? 'node.exe' : 'node'
+    def nodeScript = '''
+    const path = require('path');
+    const libModulePath = require.resolve('@datadog/mobile-react-native');
+    const packagePath = path.resolve(path.dirname(libModulePath), '../..')
+    console.log(packagePath);
+    '''
+
+    def stdout = new ByteArrayOutputStream()
+    def process = new ProcessBuilder(nodeExecutable)
+            .redirectErrorStream(true)
+            .start()
+
+    process.outputStream.withWriter { writer ->
+        writer << nodeScript
+    }
+
+    process.inputStream.eachLine { line ->
+        stdout << line << '\n'
+    }
+    
+    process.waitFor()
+
+    def resolvedPath = stdout.toString().trim()
+    if (isValidDatadogPackagePath(resolvedPath)) {
+        return resolvedPath
+    } else {
+        println("WARNING: Could not resolve @datadog/mobile-react-native path, falling back to default: ${defaultPath}")
```

Comment:
Perhaps we could also log the `resolvedPath` that was obtained after trimming so at least know what it was trying to parse and have more info when trying to fix it?